textalyze.js

```diff
@@ -74,6 +78,45 @@ function getPrintStatistics(map) {
   return output;
 }
 
+/**
+ * Given an input Map, gets the histogram text to print it to the user.
+ * @param {Map} map - The map to be printed to the user.
+ */
+function getHistogramPrintStatistics(map) {
+  if (typeof map !== 'object' || !(map instanceof Map)) {
+    throw new TypeError('The item frequencies must be a valid map.');
+  }
+
+  /**
+   * Given a percent size, creates a bar for displayal in the histogram.
+   * @param {Number} lengthPercent - The size of the bar as a percent value.
+   * @returns {String} bar - The bar string.
+   */
+  function makeHistogramBar(lengthPercent) {
+    // In order to make the bar display even with the most small frequencies, this value needs to be big.
+    const maxBarLength = 400;
```

Comment:
I think you realize this is a hack, but what happens if I do `getHistogramPrintStatistics(new Map([['a', 1]]))`?  It will break my terminal output.

We want to display the actual percentage,  but you might want the bar height to be relative to the maximum count rather than the sum of all counts.  So the most frequent letter always has a full-length bar and everything else is relative to that.

That way you can make `maxBarLength` something like `40` and it will always look reasonable.  You could also do something more dynamic, like calculate the width of the terminal window (Google for how to do that if you're interested).